src/chain.rs

```diff
@@ -59,3 +53,71 @@ impl From<BNetwork> for Network {
         }
     }
 }
+
+/// Network ID is identifier of the Tapyrus network
+#[derive(Clone, Debug, PartialEq)]
+pub struct NetworkId(u32);
```

Comment:
`Network` and `NetworkId` are separated, but isn't it better for the structure to have the mode(prod or dev) and id in the `Network` struct?